src/bus/mod.rs

```diff
@@ -75,6 +74,16 @@ impl Bus {
             _ => bail!("Unsupported address `{}`.", address),
         };
 
+        let guid: OwnedGuid = match address.guid() {
```

Comment:
why delay this? :thinking: If there is no guid in the server address, it won't be there in the client either, right?

Reply:
I was originally concerned that I was losing this when I replaced `address` with the resolved / client-facing value, which is why I swapped them

I've reverted